transport/http2_client.go

```diff
@@ -508,10 +491,16 @@ func (t *http2Client) CloseStream(s *Stream, err error) {
 		return
 	}
 	t.mu.Unlock()
-	if updateStreams {
-		t.streamsQuota.add(1)
-	}
+	var rstStream bool
+	defer func() {
+		if !rstStream {
```

Comment:
Add comments here to explain what rstStream means and our special handling for streams quota pool.